### PR TITLE
flag value bindings for kubectl apply commands

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
/kind cleanup
/sig cli

xxxOptions did not get bound as default value for some flags.
This PR cleans those flag bindings for `kubectl apply` commands.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #60366

**Special notes for your reviewer**:
/assign @deads2k 

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/kubectl/cmd/apply_view_last_applied.go [hunk 1]
```diff
@@ -140,7 +145,7 @@ func (o *ViewLastAppliedOptions) RunApplyViewLastApplied() error {
 				return err
 			}
 			fmt.Fprint(o.Out, string(jsonBuffer.Bytes()))
-		case "yaml":
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply_view_last_applied.go]: leave the original case I think.  Add a `default` that fails.

File: pkg/kubectl/cmd/apply_view_last_applied.go [hunk 2]
```diff
@@ -66,23 +66,28 @@ func NewCmdApplyViewLastApplied(f cmdutil.Factory, out, err io.Writer) *cobra.Co
...
+	cmd.Flags().BoolVar(&options.All, "all", options.All, "Select all resources in the namespace of the specified resource types")
 	usage := "that contains the last-applied-configuration annotations"
 	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
 
 	return cmd
 }
 
-func (o *ViewLastAppliedOptions) Complete(f cmdutil.Factory, args []string) error {
+func (o *ViewLastAppliedOptions) Complete(cmd *cobra.Command, f cmdutil.Factory, args []string) error {
+	if o.OutputFormat != "" {
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply_view_last_applied.go]: restructure into a switch case for me that allows empty, yaml, and json, then has this default which fails?

File: pkg/kubectl/cmd/apply.go [hunk 3]
```diff
@@ -99,7 +102,7 @@ var (
 )
 
 func NewCmdApply(baseName string, f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
-	var options ApplyOptions
+	options := ApplyOptions{Overwrite: true, Cascade: true, GracePeriod: -1, OpenApiPatch: true}
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply.go]: same comment about `NewApplyOptions() *ApplyOptions` here.

**Comment by deads2k** [comment]
These changes are really good. 

/assign @soltysh 
/assign @juanvallejo

File: pkg/kubectl/cmd/apply_view_last_applied.go [hunk 4]
```diff
@@ -140,29 +139,19 @@ func (o *ViewLastAppliedOptions) RunApplyViewLastApplied() error {
 				return err
 			}
 			fmt.Fprint(o.Out, string(jsonBuffer.Bytes()))
-		case "yaml":
+		case "yaml", "":
```
Comment (soltysh) [inline pkg/kubectl/cmd/apply_view_last_applied.go]: Why we have `""`?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/kubectl/cmd/apply_view_last_applied.go (hunk 4 above)
Comment (dixudx) [reply pkg/kubectl/cmd/apply_view_last_applied.go]: Just in case the value will be an empty string.

Comment (dixudx) [reply pkg/kubectl/cmd/apply_view_last_applied.go]: And this is modified based on the old method `ValidateOutputArgs`, where `""` is counted as well.

Comment (juanvallejo) [reply pkg/kubectl/cmd/apply_view_last_applied.go]: Looks like we default to yaml on empty string, mind doing a check before the switch (maybe in the Complete method) that defaults the value of `o.OutputFormat` to `yaml` if empty?

**Comment by dixudx** [comment]
@juanvallejo @soltysh Updated. PTAL. Thanks.

**Review by soltysh — APPROVED** [review]

**Pull request merged**